apiserver/facades/client/machinemanager/machinemanager.go

```diff
@@ -271,26 +271,41 @@ func (mm *MachineManagerAPI) destroyMachine(args params.Entities, force, keep bo
 	if err := mm.check.RemoveAllowed(); err != nil {
 		return params.DestroyMachineResults{}, err
 	}
-	destroyMachine := func(entity params.Entity) (*params.DestroyMachineInfo, error) {
+	destroyMachine := func(entity params.Entity) params.DestroyMachineResult {
+		result := params.DestroyMachineResult{}
+		fail := func(e error) params.DestroyMachineResult {
+			result.Error = common.ServerError(e)
+			return result
+		}
+
 		machineTag, err := names.ParseMachineTag(entity.Tag)
 		if err != nil {
-			return nil, err
+			return fail(err)
 		}
 		machine, err := mm.st.Machine(machineTag.Id())
 		if err != nil {
-			return nil, err
+			return fail(err)
 		}
 		if keep {
 			logger.Infof("destroy machine %v but keep instance", machineTag.Id())
 			if err := machine.SetKeepInstance(keep); err != nil {
-				return nil, err
+				if !force {
+					return fail(err)
+				}
+				result.Error = common.ServerError(err)
```

Comment:
Surely if we are forcing we don't want an error in result.Error? That is what fail is doing right?

Reply:
This error does not prevent the actual destroy call to take place. This is just a returned result...